torch/_dynamo/bytecode_analysis.py

```diff
@@ -132,46 +133,52 @@ class FixedPointBox:
 class StackSize:
     low: Real
     high: Real
-    fixed_point: FixedPointBox
 
     def zero(self):
         self.low = 0
         self.high = 0
-        self.fixed_point.value = False
 
-    def offset_of(self, other, n):
+    def offset_of(self, other, n, fixed_point):
         prior = (self.low, self.high)
         self.low = min(self.low, other.low + n)
         self.high = max(self.high, other.high + n)
         if (self.low, self.high) != prior:
-            self.fixed_point.value = False
+            fixed_point.value = False
+        else:
+            fixed_point.value = True
 
 
 def stacksize_analysis(instructions):
     assert instructions
+
     fixed_point = FixedPointBox()
     stack_sizes = {
-        inst: StackSize(float("inf"), float("-inf"), fixed_point)
-        for inst in instructions
+        inst: StackSize(float("inf"), float("-inf")) for inst in instructions
     }
     stack_sizes[instructions[0]].zero()
 
-    for _ in range(100):
-        if fixed_point.value:
-            break
-        fixed_point.value = True
+    indexof = {id(inst): i for i, inst in enumerate(instructions)}
 
-        for inst, next_inst in zip(instructions, instructions[1:] + [None]):
-            stack_size = stack_sizes[inst]
-            if inst.opcode not in TERMINAL_OPCODES:
-                assert next_inst is not None, f"missing next inst: {inst}"
-                stack_sizes[next_inst].offset_of(
-                    stack_size, stack_effect(inst.opcode, inst.arg, jump=False)
-                )
-            if inst.opcode in JUMP_OPCODES:
-                stack_sizes[inst.target].offset_of(
-                    stack_size, stack_effect(inst.opcode, inst.arg, jump=True)
-                )
+    worklist = list()
```

Comment:
Suggest using `collections.deque()`, the `pop(0)` below is `O(n)` with a list, `O(1)` with a deque.

Reply:
Thanks for your advice! I will change it accordingly.